controllers/volumereplicationgroup_controller.go

```diff
@@ -934,27 +1018,38 @@ func (v *VRGInstance) validateS3Endpoint(s3Endpoint, s3Bucket string) error {
 // processVRAsPrimary processes VR to change its state to primary, with the assumption that the
 // related PVC is prepared for VR protection
 func (v *VRGInstance) processVRAsPrimary(vrNamespacedName types.NamespacedName, log logr.Logger) error {
-	return v.createOrUpdateVR(vrNamespacedName, volrep.Primary, log)
+	return v.createOrUpdateVR(vrNamespacedName, ramendrv1alpha1.Primary, log)
 }
 
 // processVRAsSecondary processes VR to change its state to secondary, with the assumption that the
 // related PVC is prepared for VR as secondary
 func (v *VRGInstance) processVRAsSecondary(vrNamespacedName types.NamespacedName, log logr.Logger) error {
-	return v.createOrUpdateVR(vrNamespacedName, volrep.Secondary, log)
+	return v.createOrUpdateVR(vrNamespacedName, ramendrv1alpha1.Secondary, log)
 }
 
 // createOrUpdateVR updates an existing VR resource if found, or creates it if required
-// TODO: We need to ensure VR is in the desired state and requeue if not
+// While both creating and updating the VolumeReplication resource, conditions.status
+// for the protected PVC (corresponding to the VolumeReplication resource) is set as
+// PVCProgressing. When the VolumeReplication resource changes its state either due to
+// successful reaching of the desired state or due to some error, VolumeReplicationGroup
+// would get a reconcile. And then the conditions for the appropriate Protected PVC can
+// be set as either Replicating or Error.
 func (v *VRGInstance) createOrUpdateVR(vrNamespacedName types.NamespacedName,
```

Comment:
While I agree with this, just want to clarify another thought I have.

Do we want to minimize the places (or functions) which knows about VolumeReplication states? Like with this change (as it is in this PR currently), only createOrUpdateVR function knows about VolumeReplication state and the rest of the VolumeReplicationGroup code need not know about the conversion and leave to the lowest function where all the necessary conversions happen. It keeps these conversions (and the knowledge of VRG's state and VR's state being separate entities) to only createOrUpdateVR function (and the helper function it calls such as updateVR).

Let me know your thoughts on this.

Reply:
So the code from `processVRAsPrimary/Secondary` is already an VR function, so starting from this layer the functions processing `VR.State` instead of `Ramen.State` would be better.